examples/dtls-echo/dtls-client.c

```diff
@@ -93,48 +74,89 @@ static const unsigned char ecdsa_pub_key_y[] = {
     0xE9, 0x3F, 0x98, 0x72, 0x09, 0xDA, 0xED, 0x0B,
     0x4F, 0xAB, 0xC3, 0x6F, 0xC7, 0x72, 0xF8, 0x29
 };
+#endif
 
+/* TinyDTLS callback for detecting the state of the DTLS channel. */
+static int _events_handler(struct dtls_context_t *ctx,
+                           session_t *session,
+                           dtls_alert_level_t level,
+                           unsigned short code)
+{
+    (void) ctx;
+    (void) session;
+    (void) level;
 
-/**
- * @brief This care about getting messages and continue with the DTLS flights.
- * This will handle all the packets arriving to the node.
- * Will determine if its from a new DTLS peer or a previous one.
+    if (code == DTLS_EVENT_CONNECTED) {
+        dtls_connected = 1;
+        DEBUG("CLIENT: DTLS Channel established!\n");
+    }
+#if ENABLE_DEBUG == 1
+    /* At least a DTLS Client Hello was prepared? */
+    else if (code == DTLS_EVENT_CONNECT) {
+        DEBUG("CLIENT: DTLS Channel started\n");
+    }
+#endif
+
+    /*TODO: DTLS_EVENT_RENEGOTIATE */
```

Comment:
Still a todo ?

Reply:
For this example, yes.  Though is not truly required and I'm not sure if could contribute in giving information to other developers.   However, is very helpful for asynchronous events.